dj_hetmech_app/views.py

```diff
@@ -64,13 +63,23 @@ def polish_pathcounts(self, source_id, target_id, pathcounts_data):
         return pathcounts_data
 
     def get(self, request):
+        # Validate "source" parameter
         source_id = request.query_params.get('source', None)
-        target_id = request.query_params.get('target', None)
+        if source_id is None:
+            return Response({'error': 'source parameter not found in URL.'})
         try:
             source_node = Node.objects.get(pk=source_id)
+        except:
+            return Response({'error': 'source node not found in database.'})
+
+        # Validate "target" parameter
+        target_id = request.query_params.get('target', None)
```

Comment:
Is `query_params` a dict? If so None is default value for get. Still okay to include if want to be explicit.

Reply:
Yes, I want to be explicit here.